daffodil-test/src/test/resources/org/apache/daffodil/section23/dfdl_expressions/expressions_unused_path_no_context.dfdl.xsd

```diff
@@ -28,20 +28,119 @@
     </xs:appinfo>
   </xs:annotation>
 
+
   <xs:element name="e1" type="xs:int" dfdl:lengthKind="explicit" dfdl:length="1" />
 
-  <xs:group name="unused_group">
+  <xs:group name="unused_group_1">
     <xs:sequence>
       <xs:element name="value1" type="xs:string" dfdl:lengthKind="explicit" dfdl:length="1" />
       <xs:sequence dfdl:terminator="{ ./value1 }" />
```

Comment:
So, this is a bit odd. The . in ./value1 refers to "some parent element" because this is inside a group. 

But we don't care what element that is. 

I'm ok with this issuing a warning, I think, so long as existing schemas don't pour out tons of such warnings. 
But this is the borderline case where it would be nice if there was a way to avoid the warning in this case. 

I can see how the expression compiler can't cope with this though. It wants that Up step to refer to an element that has DPathCompileInfo. There is no such. 

So this is a corner case where we definitely cannot compile the expression, but we may not want to issue a warning. 

I'd say let's merge this in, and try  it out and if we get too many such warnings deal with it then.

Reply:
Yep, that's exactly what's going on. Similar to my other comment, when we compile unused groups we could maybe create a fake "root" element contain a group ref to the unused group. This way `./` would refer to something with a DPathElementCompileInfo and it would work as expected. And if it did actually go outside the fake root, only then would you get a warning. That could potentially reduce a lot of warnings if it's something we did start to see more of them.